django/db/models/fields/__init__.py

```diff
@@ -2792,14 +2792,24 @@ def __init__(self, *args, **kwargs):
     def check(self, **kwargs):
         return [
             *super().check(**kwargs),
-            *self._check_primary_key(),
+            *self._check_primary_key(kwargs.get("databases", ["default"])),
         ]
 
-    def _check_primary_key(self):
-        if not self.primary_key:
+    def _check_primary_key(self, databases):
+        if not (
+            self.primary_key
+            or (
+                all(
+                    connections[db].features.supports_autofields_in_composite_pk
+                    for db in databases
+                )
```

Comment:
Simon had recently mentioned that only checking the default connection's features was harmful in a multi-engine setup.  Is this what we want to do moving forward or just continue to use the default connection?  If so we may need to abstract this as it will catch folks out.